felix/bpf/polprog/pol_prog_builder.go

```diff
@@ -94,9 +94,14 @@ const (
 
 var (
 	// Stack offsets.  These are defined locally.
-	offStateKey    = nextOffset(4, 4)
-	offSrcIPSetKey = nextOffset(ipsets.IPSetEntrySize, 8)
-	offDstIPSetKey = nextOffset(ipsets.IPSetEntrySize, 8)
+	offStateKey = nextOffset(4, 4)
+	// Even for v4 progrtams, we have the v6 layout
+	// N.B. we can use 64bit stack stores in ipv6 because of the 4 byte alignment
+	// preceded by the 4 bytes of StateKey. That makes the ip within the src key
+	// 8 bytes aligned. And since the ip is followed by 4 bytes of
+	// port+proto+pad, the dst key is also aligned in the same way. <sweat :-)>
```

Comment:
```suggestion
	// port+proto+pad, the dst key is also aligned in the same way. <sweet :-)>
```
Can you assert on this somehow so that we'll fail to compile if we accidentally break alignment?